parsely/src/androidTest/java/com/parsely/parselyandroid/FunctionalTests.kt

```diff
@@ -70,6 +71,47 @@ class FunctionalTests {
         }
     }
 
+    /**
+     * In this scenario, the consumer app tracks 2 events during the first flush interval.
+     * Then, we validate, that after flush interval passed the SDK sends the events
+     * to Parse.ly servers.
+     *
+     * Then, the consumer app tracks another event and we validate that the SDK sends the event
+     * to Parse.ly servers as well.
+     */
+    @Test
+    fun appFlushesEventsAfterFlushInterval() {
+        ActivityScenario.launch(SampleActivity::class.java).use { scenario ->
+            scenario.onActivity { activity: Activity ->
+                beforeEach(activity)
+                server.enqueue(MockResponse().setResponseCode(200))
+                parselyTracker = initializeTracker(activity)
+
+                parselyTracker.trackPageview("url", null, null, null)
+            }
+
+            Thread.sleep((flushInterval / 2).inWholeMilliseconds)
+
+            scenario.onActivity {
+                parselyTracker.trackPageview("url", null, null, null)
+            }
+
+            Thread.sleep((flushInterval / 2).inWholeMilliseconds)
+
+            val firstRequestPayload = server.takeRequest(2000, TimeUnit.MILLISECONDS)?.toMap()
```

Comment:
Such a long waiting time is needed because of poor performance on GitHub Actions emulator. Also, the tests are executed differently, comparing to local execution (both device and same-settings emulator) and Firebase Test Lab. You can compare the logs, that just after `[Parsely] Sending request with 2 events` log, GitHub Actions emulator sends test activity to `Paused` (`Lifecycle status change: com.parsely.parselyandroid.FunctionalTests$SampleActivity@316b36c in: PAUSED`), when it doesn't happen anywhere else...

All of this makes me think about pivoting to:
- using FIrebase Test Lab instead CI emulator
- moving from emulator-based functional test to Robolectric

Reply:
> using FIrebase Test Lab instead CI emulator

Hmmm, maybe, not sure? 🤔 

> moving from emulator-based functional test to Robolectric

Hmmm, if I could vote, I would say to NOT making that move to Robolectric. 😊